src/M6Web/Bundle/RedisBundle/CacheAdapters/RedisCacheItemPoolAdapter.php

```diff
@@ -0,0 +1,185 @@
+<?php
+namespace M6Web\Bundle\RedisBundle\CacheAdapters;
+
+use M6Web\Bundle\RedisBundle\Redis\Redis as BaseRedis;
+
+use Symfony\Component\Cache\CacheItem;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheItemInterface;
+
+/**
+ * Class RedisCacheItemPoolAdapter
+ */
+class RedisCacheItemPoolAdapter extends BaseRedis implements CacheItemPoolInterface
+{
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    /**
+     * @var \Closure
+     */
+    private $getItemLifeTime;
+
+    /**
+     * @var bool
+     */
+    protected $transactionInProgress = false;
+
+    public function __construct($redis, $defaultLifetime = 0)
+    {
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit, $cacheExpire) use ($defaultLifetime) {
+                $cacheItem = new CacheItem();
```

Comment:
Why don't we subclass `CacheItem`, instead of using closure binding to access protected properties?